File: mobile/src/types/student.ts
export type HistoryTrend = "up" | "down" | "stable";

export type StudentHistoryItem = {
  attempt_id: number;
  test_type: "diagnostic" | "adaptive" | "final";
  started_at: string;
  finished_at: string;
  score_percent: number;
  level_result: string;
  theta_final: number;
};

export type StudentHistorySummary = {
  total_attempts: number;
  average_score: number;
  best_score: number;
  trend: HistoryTrend;
};

export type StudentHistoryResponse = {
  summary: StudentHistorySummary;
  attempts: StudentHistoryItem[];
};

File: backend/app/schemas/student.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel

from app.db.models import TestType


class StudentHistoryItem(BaseModel):
    attempt_id: int
    test_type: TestType
    started_at: datetime
    finished_at: datetime
    score_percent: float
    level_result: str
    theta_final: float


class StudentHistorySummary(BaseModel):
    total_attempts: int
    average_score: float
    best_score: float
    trend: str


class StudentHistoryResponse(BaseModel):
    summary: StudentHistorySummary
    attempts: list[StudentHistoryItem]
